### ISSUE TITLE
update ConfigurationMapFactoryBean to accept aditional map

### ISSUE THREAD

**Issue by kofemann** [dCache/dcache#4625]
The ConfigurationMapFactoryBean used to collect configuration properties for a spring bean from dcache environment by selecting keys based on specified prefix, like `some.prefix!`. This is used to inject properties into 3rd party components, like BerkeleyDB or Apache Kafka producer. However, in some situations we need in addition to prefixed keys enforce other values as well.

Update ConfigurationMapFactoryBean to allow enforcing  some values by providing a Ma with initial values.

```property
some.client.config!client.option=value
```

```xml
  <bean id="client-config"
      class="org.dcache.util.configuration.ConfigurationMapFactoryBean">
    <property name="prefix" value="some.client.config"/>
    <property name="initialProperties">
          <map>
              <entry key="some-key"  value="some value" />
          </map>
    </property>
  </bean>
```

**Comment by paulmillar** [comment]
Could you provide some concrete examples?

**Comment by kofemann** [comment]
Real-life use case: kafka brokers vs kafka producer options:
```property
# this must be set by admin
dcache.kafka.brokers=broker1,broker2

# this can be set based on desired kafka cnfiguration
kafka.config!metrics.sample.window.ms=1000
...
```

**Comment by paulmillar** [comment]
Wouldn't this be satisfied by:
```
(deprecated)dcache.kafka.brokers=...
kafka.config!bootstrap.servers=${dcache.kafka.brokers}
```
Later, once sites have migrated, remove the `dcache.kafka.brokers` property.

**Comment by kofemann** [comment]
The point is to keep dcache relevant configuration without prefix

**Comment by paulmillar** [comment]
Sorry, I still don't understand.

The `dcache.kafka.brokers` property (currently `dcache.kafka.bootstrap-servers`) is only used by the embedded Kafka client, not by any dCache code.

It still unclear to me what is the benefit of the `dcache.kafka.brokers` property over having all the kafka client configuration with a prefix; e.g.,
```
kafka.config!bootstrap.servers=...
```

The original message mentions supporting default values.

I believe this is already available by including the default value in the regular defaults file; e.g., an entry like `kafka.config!bootstrap.servers = localhost:9092` in `skel/share/defaults/dcache.properties`.

**Comment by paulmillar** [comment]
Resolved with commit f87e4a07.
